h2o-core/src/test/java/water/rapids/ast/prims/PermutationFeatureImportanceTest.java

```diff
@@ -0,0 +1,328 @@
+import org.junit.Test;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+import water.rapids.*;
+import water.rapids.vals.ValFrame;
+import water.util.*;
+import org.junit.BeforeClass;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+import static hex.genmodel.utils.DistributionFamily.gaussian;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class PermutationFeatureImportanceTest extends TestUtil {
+
+    @BeforeClass
+    public static void stall() {
+        stall_till_cloudsize(1);
+    }
+    
+    @Test
+    public void shuffleSmallDoubleVec(){
+        Frame t = null;
+        try {
+            Scope.enter();
+            t = new TestFrameBuilder()
+                    .withVecTypes(Vec.T_NUM, Vec.T_NUM)
+                    .withColNames("first", "second")
+                    .withDataForCol(0, ard(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0))
+                    .withDataForCol(1, ard(1.1, 2.1, 3.1, 4.1, 5.1, 6.1, 7.1, 8.1))
+                    .build();
+            Scope.track(t);
+            Vec OG_vec = t.vec("first").doCopy();
+//            VecUtils.shuffleVec new_vv = new VecUtils.shuffleVec(t.vec(0)).doAll(t.vec("first"));
+            Vec new_v = Vec.makeVec(new VecUtils.shuffleVecVol2().doAll(t.vec("first"))._result, Vec.newKey());
+
+//            Vec new_v = new_vv._vec;
+            double rnd_coe = randomnessCoeefic(OG_vec, new_v);
+
+            Assert.assertNotEquals(0, rnd_coe);
+            
+        } finally {
+            Scope.exit();
+        }
+        
+    }
+
+    // compares values of vec and returns the randomness change. 0 no change, 1.0 all rows shuffled
+    double randomnessCoeefic(Vec og, Vec nw){
+        int changed_places = 0;
+        for (int i = 0; i < nw.length() ; ++i){
+            if(og.at(i) != nw.at(i))
+                changed_places++;
+        }
+        return changed_places * 1.0/nw.length();
+    }
+    
+    public double [] randomnessCoef(Frame fr, double [] res){
+        // copy vecs to check on permuated ones
+        Vec OG_vec[] = new Vec[fr.numCols()];
+        for (int i = 0 ; i <  fr.numCols(); i ++)
+            OG_vec[i] = fr.vec(i).makeCopy();
+
+        String [] features = fr.names();
+        ShuffleVec shf = new ShuffleVec(fr);
+        Vec shf_v;
+        for (int i = 0 ; i <  fr.numCols(); i ++){
+            String col_name = features[i];
+            shf_v = ShuffleVec.ShuffleTask.shuffle(fr.vec(col_name));
+            res[i] = randomnessCoeefic(OG_vec[i] , shf_v);
+        }
+        return res;
+    }
+    
+    @Test
+    public void shuffleVecBiggerData(){
+        Frame fr = null;
+        try {
+            Scope.enter(); 
+            fr = parse_test_file("./smalldata/airlines/AirlinesTrainWgt.csv"); // has 24422; should be enough to test speed 
```

Comment:
This file is not so big to test performance let alone MRTask is working. The number of rows is so small to create more than one chunk.